endpoint.go

```diff
@@ -366,16 +365,11 @@ func (ep *endpoint) Delete() error {
 	driver := n.driver
 	delete(n.endpoints, epid)
 	n.Unlock()
-	defer func() {
-		if err != nil {
-			n.Lock()
-			n.endpoints[epid] = ep
-			n.Unlock()
-		}
-	}()
 
-	err = driver.DeleteEndpoint(nid, epid)
-	return err
+	if err := driver.DeleteEndpoint(nid, epid); err != nil {
```

Comment:
As the commit log says, I feel we should check for FrobiddenError here as well.
We should not assume driver will never send a forbidden error for endpoint delete only because we know that actual driver implementation does not throw one.